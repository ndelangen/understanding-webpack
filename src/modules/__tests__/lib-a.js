import * as a from '../lib-a';

describe('sanity', () => {
  it('should exist', () => {
    expect(a).toBeDefined();
  });
  it('should contain the right methods', () => {
    expect(Object.keys(a)).toMatchSnapshot();
  });
});

describe('functional', () => {
  describe('MY_CONST', () => {
    it('should be defined', () => {
      expect(a.MY_CONST).toBeDefined();
    });
  });

  describe('default', () => {
    it('should be defined', () => {
      expect(a.default).toBeDefined();
    });
  });

  describe('default', () => {
    it('should be defined', () => {
      expect(a.default).toBeDefined();
    });
  });

  describe('myFunc', () => {
    it('should not throw', () => {
      expect(a.myFunc).not.toThrow();
    });
    it('should return MY_CONST', () => {
      expect(a.myFunc()).toBe(a.MY_CONST);
    });
  });

  describe('myArrowFunc', () => {
    it('should not throw', () => {
      expect(a.myArrowFunc).not.toThrow();
    });
    it('should return MY_CONST', () => {
      expect(a.myArrowFunc()).toBe(a.MY_CONST);
    });
  });

  describe('MyClass', () => {
    it('should return an instance with property `foo`', () => {
      expect(new a.MyClass()).toEqual({ foo: a.MY_CONST });
    });
  });
});
