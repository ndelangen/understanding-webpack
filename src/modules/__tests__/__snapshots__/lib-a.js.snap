// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sanity should contain the right methods 1`] = `
Array [
  "myFunc",
  "default",
  "MY_CONST",
  "myArrowFunc",
  "MyClass",
]
`;
